### PR TITLE
Redesign search bar & config display

### PR DESCRIPTION
**Pull request by Adebesin-Cell**
## Redesign the search bar & config display

**Changes:**
- Added new search bar layout
- Worked UI to show multiple sources from ``MatchCount``
- Reduced character limit to 1000
- Fixes ``MatchCount`` default to 1
- Added ``enter`` event listener to the textarea.
- Made textarea resize based on user typing.

fixes https://github.com/EveripediaNetwork/issues/issues/1315
fixes https://github.com/EveripediaNetwork/issues/issues/1381

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -99,6 +113,8 @@ export const getServerSideProps: GetServerSideProps = getLocaleProps(
     });
     await prisma.$disconnect();
 
+    console.log(cachedRequest);
```
Comment (Royal-lobster) [inline src/pages/index.tsx:116]:
```suggestion
```

File: src/components/Search/index.tsx [hunk 2]
```diff
@@ -53,26 +59,35 @@ const Search = ({ initialQuery, debugOptions, cache }: SearchProps) => {
...
+  const hasMultipleSources = debugOptions.matchCount > 1;
+
   const isFetchingAnswer =
     !isUsingCachedQuery && (isFetching || (generating && !answer));
 
   const sources = isUsingCachedQuery
     ? cache.pages.map((p) => ({ title: p.title, url: p.url }))
     : information?.chunks.map((c) => ({ title: c.title, url: c.url }));
 
+  console.log(sources);
```
Comment (Royal-lobster) [inline src/components/Search/index.tsx:71]:
```suggestion
```

**Comment by Softdev1** [comment]
Can we update on this @Adebesin-Cell

**Review by Simplypheyie96 — CHANGES_REQUESTED** [review]
- since max is 10, the match count shouldn't go beyond that.
Screenshot 2023-05-31 at 11 40 56 AM
- For links, make them pink on hover
image

**Comment by Tolu-Mals** [comment]
The font size for the other wiki sources looks small. It's meant to have the same font size as the original source.

**Comment by Tolu-Mals** [comment]
The font in the design for the 'more sources' button is 'Inconsolata', but you used 'Montserrat' here

**Comment by Simplypheyie96** [comment]
> The font in the design for the 'more sources' button is 'Inconsolata', but you used 'Montserrat' here

i think there is incosistencies with these things. i'll tell wale to work on that.
the primary link is the one that should have a different typeface. every other one should be montserrat

**Comment by Adebesin-Cell** [comment]
Good catch 👏 @Simplypheyie96 @Tolu-Mals, I'll add the fixes.

**Comment by Tolu-Mals** [comment]
We should remove the focus style. Also the text area should be bigger by default, the way it was previously.
Screenshot 2023-05-31 at 2 42 48 PM

**Comment by Tolu-Mals** [comment]
Also it'll be nice if the search bar is focused after the page loads

**Review by Royal-lobster — APPROVED** [review]

**Pull request merged**